### ISSUE TITLE
Investigate whether there is a way to use sparse target for multilabel classification in tensorflow

### ISSUE THREAD

**Issue by nsorros** [wellcometrust/grants_tagger#111]
Since the volume of the targets are quite big in grants_tagger we use sparse labels to reduce the memory footprint. This creates a problem when training neural nets using tensorflow because as far as I know I have not found a way to pass this sparse vector directly so the way around is that as data are being batched I convert Y to dense. Example
https://github.com/wellcometrust/grants_tagger/blob/720bdc46ff3f7b09407d8d01825450a67742bad6/grants_tagger/models.py#L141

Apart from the above examples, there are a couple of other places where we deal with that and it would be much more convenient to be able to pass the sparse vector directly

**Comment by nsorros** [comment]
no use of tensorflow at this point only torch
